### ISSUE TITLE
Lightning failing with 'Ran out of routes to try after 1 attempt: see paystatus'

### ISSUE THREAD

**Issue by nachikettapas** [ElementsProject/lightning#3462]
When initiating multiple pay commands (one after another), the lightning fails with error 'Ran out of routes to try after 1 attempt: see paystatus'

lightningClient.pay(invoice)
lightningClient.pay(invoice)

If user only once, then it is working. 

ubuntu@iot-1:~$ lightning/cli/lightning-cli --network=regtest getinfo
{
   "id": "027453cb5c7475208110db2366a8590b866135095c2ba98ac8a1e79c5394cd4d9f",
   "alias": "ORANGEMASTER",
   "color": "027453",
   "num_peers": 1,
   "num_pending_channels": 0,
   "num_active_channels": 1,
   "num_inactive_channels": 0,
   "address": [],
   "binding": [
      {
         "type": "ipv6",
         "address": "::",
         "port": 9735
      },
      {
         "type": "ipv4",
         "address": "0.0.0.0",
         "port": 9735
      }
   ],
   "version": "v0.8.0-124-g038ee28",
   "blockheight": 240,
   "network": "regtest",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat",
   "lightning-dir": "/home/ubuntu/.lightning/regtest"
}

**Comment by s-tikhomirov** [comment]
May be indirectly related to #3463 (which, in a coincidence, was created 30 minutes after this one).

**Comment by nachikettapas** [comment]
Well, it was working till yesterday. The same code is not working now. Doesn't seem related but I will have a look.

**Comment by cdecker** [comment]
Can you share the channel capacities and the amount you are trying to send through? It is likely that your channels just don't have enough capacity to send the amount through

**Comment by nachikettapas** [comment]
The following is the steps I performed.
1. I transferred 5 bitcoin from bitcoin wallet to lightning wallet.
2. I created and funded a channel with max fund 16777215 satoshis.
3. Then I create a invoice for 1msatoshi multiple times. 

I get the following error:
```
2020-01-30T18:53:24.385Z iot Decoded invoice for piece index = 6 from peerId = 2d5757303030312d6a426a395a637452495a7142 is { currency: 'bcrt', created_at: 1580410404, expiry: 604800, payee: '024989ef625965b5b516e05ede377d5eed9777257db26ebaddf0442e22927d0d98', msatoshi: 1, amount_msat: '1msat', description: 'transaction', min_final_cltv_expiry: 10, payment_secret: 'e4cdaf4365d9ed32f88f5654acc6addc4d254b94363ac9b71c97ce2dc2e1e018', features: '028200', payment_hash: 'd2a29b9f112a4972dca72a33bf16a4255ca34cd42e746cda3bfe7fae2f9ce5c9', signature: '3045022100b95de8980bd6593bbb25d6f8689d7b1dd516945dc4714fe998cf9d0293594ec402207174a68e3fcbc43130c7ba6e95210f1dc10aafd7bdafc19da62a69b80c194742' }
IoT: paying for piece 6, for node 024989ef
2020-01-30T18:53:24.633Z LightningClient err = { error: { code: 210, message: 'Ran out of routes to try after 1 attempt: see paystatus' }, stack: 'Error\n    at LightningClient.call (/home/ubuntu/patching-lightning/node_modules/lightning-client/index.js:139:27)\n    at LightningClient.(anonymous function) [as pay] (/home/ubuntu/patching-lightning/node_modules/lightning-client/index.js:175:21)\n    at LightningClient.pay (/home/ubuntu/patching-lightning/Utils/LightningClient.js:142:38)\n    at IoT.decodeAndPay (/home/ubuntu/patching-lightning/IoT/IoT.js:151:49)\n    at process._tickCallback (internal/process/next_tick.js:68:7)' }
```

**Comment by darosior** [comment]
You (usually) need to compensate routing nodes for their work, so nobody would transfer 1msat (the smaller unit transferable) as they could not take fee.

Try with a hundred of sats ! :-)

**Comment by ZmnSCPxj** [comment]
You will have to use `paystatus` command for more detail on the failure.  It looks like you are using a different UI; you might want to file an issue with them as well, so that they can actually use `paystatus` to get details about the failure.  `paystatus` will give you a full report of the log of the payment.

**Comment by nachikettapas** [comment]
Sure I will give it a try and I understand the business aspect of it. 
What I don't understand is why it was working a few days back and all of a sudden the same setup is generating error?
Meanwhile, I am also trying older commits to see if it works.

**Comment by ZmnSCPxj** [comment]
Can you describe the setup in more detail?  How many nodes, what channels are set up, etc.?

**Comment by cdecker** [comment]
> Meanwhile, I am also trying older commits to see if it works.

It is unlikely that older c-lightning versions will work, since the problem appears not to be with c-lightning, but rather the channel configuration that you are using to test.

**Comment by nachikettapas** [comment]
The following is the setup I am trying.
1. There are 3 nodes A, B and C
2. There is a channel between A and B (capacity 16777215sat).
3. Similarly there is a channel between A and C (capacity 16777215sat).
4. Now I am trying a transaction of 100sat from B to C which is failing with "Ran out of routes".

```
ubuntu@bitcoin-node-1:~$ lightning/cli/lightning-cli --network=regtest listchannels
{
   "channels": [
      {
         "source": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
         "destination": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
         "short_channel_id": "209x1x0",
         "public": true,
         "satoshis": 16777215,
         "amount_msat": "16777215000msat",
         "message_flags": 1,
         "channel_flags": 0,
         "active": true,
         "last_update": 1580467310,
         "base_fee_millisatoshi": 1,
		 "fee_per_millionth": 10,
         "delay": 6,
         "htlc_minimum_msat": "0msat",
         "htlc_maximum_msat": "4294967295msat"
      },
      {
         "source": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
         "destination": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
         "short_channel_id": "209x1x0",
         "public": true,
         "satoshis": 16777215,
         "amount_msat": "16777215000msat",
         "message_flags": 1,
         "channel_flags": 1,
         "active": true,
         "last_update": 1580467310,
         "base_fee_millisatoshi": 1,
         "fee_per_millionth": 10,
         "delay": 6,
         "htlc_minimum_msat": "0msat",
         "htlc_maximum_msat": "4294967295msat"
      },
      {
         "source": "02d77443ec4cb715251061b29be98cb347af453d5961227e44b7cfb6458d477331",
         "destination": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
         "short_channel_id": "221x1x0",
         "public": true,
         "satoshis": 16777215,
         "amount_msat": "16777215000msat",
         "message_flags": 1,
         "channel_flags": 0,
         "active": true,
         "last_update": 1580472464,
         "base_fee_millisatoshi": 1,
		 "fee_per_millionth": 10,
         "delay": 6,
         "htlc_minimum_msat": "0msat",
         "htlc_maximum_msat": "4294967295msat"
      },
      {
         "source": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
         "destination": "02d77443ec4cb715251061b29be98cb347af453d5961227e44b7cfb6458d477331",
         "short_channel_id": "221x1x0",
         "public": true,
         "satoshis": 16777215,
         "amount_msat": "16777215000msat",
         "message_flags": 1,
         "channel_flags": 1,
         "active": true,
         "last_update": 1580470903,
         "base_fee_millisatoshi": 1,
         "fee_per_millionth": 10,
         "delay": 6,
         "htlc_minimum_msat": "0msat",
         "htlc_maximum_msat": "4294967295msat"
      }
   ]
}
```

**Comment by cdecker** [comment]
> The following is the setup I am trying.
> 
> 1. There are 3 nodes A, B and C
> 2. There is a channel between A and B (capacity 16777215sat).
> 3. Similarly there is a channel between A and C (capacity 16777215sat).
> 4. Now I am trying a transaction of 100sat from B to C which is failing with "Ran out of routes".

Can you share the output of `listpeers` from node `A`? If `B` has no outgoing capacity on the edge `B -> A`, or `A` has no outgoing capacity on the edge `A -> C` that could be the cause for the failure.

**Comment by nachikettapas** [comment]
The listpeers output is following
```
ubuntu@bitcoin-node-1:~$ lightning/cli/lightning-cli --network=regtest listpeers
{
   "peers": [
      {
         "id": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
         "connected": true,
         "netaddr": [
            "10.71.230.72:40324"
         ],
         "globalfeatures": "",
         "localfeatures": "02aaa2",
         "features": "02aaa2",
         "channels": [
            {
               "state": "CHANNELD_NORMAL",
               "scratch_txid": "d91c930a0469cbd7698a5f06ef8fb437e9ca7b2845589af208d8e410ac6e3809",
               "owner": "channeld",
               "short_channel_id": "209x1x0",
               "direction": 1,
               "channel_id": "a67a3dc07710a639c7c20ffd5dc73358350a8e3d4122ffef3dc284ad71ccee02",
               "funding_txid": "02eecc71ad84c23defff22413d8e0a355833c75dfd0fc2c739a61077c03d7aa6",
               "close_to_addr": "bcrt1qets2l4zsfk5upwy35ml35fpjqx60zr0x3y4dcj",
               "close_to": "0014cae0afd4504da9c0b891a6ff1a243201b4f10de6",
               "private": false,
               "funding_allocation_msat": {
                  "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52": 0,
                  "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437": 16777215000
               },
               "funding_msat": {
                  "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52": "0msat",
                  "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437": "16777215000msat"
               },
               "msatoshi_to_us": 0,
               "to_us_msat": "0msat",
               "msatoshi_to_us_min": 0,
               "min_to_us_msat": "0msat",
               "msatoshi_to_us_max": 0,
               "max_to_us_msat": "0msat",
               "msatoshi_total": 16777215000,
               "total_msat": "16777215000msat",
               "dust_limit_satoshis": 546,
               "dust_limit_msat": "546000msat",
               "max_htlc_value_in_flight_msat": 18446744073709551615,
               "max_total_htlc_in_msat": "18446744073709551615msat",
               "their_channel_reserve_satoshis": 167772,
               "their_reserve_msat": "167772000msat",
               "our_channel_reserve_satoshis": 167772,
               "our_reserve_msat": "167772000msat",
               "spendable_msatoshi": 0,
               "spendable_msat": "0msat",
               "htlc_minimum_msat": 0,
               "minimum_htlc_in_msat": "0msat",
               "their_to_self_delay": 6,
               "our_to_self_delay": 6,
               "max_accepted_htlcs": 483,
               "status": [
                  "CHANNELD_NORMAL:Reconnected, and reestablished.",
                  "CHANNELD_NORMAL:Funding transaction locked. Channel announced."
               ],
               "in_payments_offered": 0,
               "in_msatoshi_offered": 0,
               "in_offered_msat": "0msat",
               "in_payments_fulfilled": 0,
               "in_msatoshi_fulfilled": 0,
               "in_fulfilled_msat": "0msat",
               "out_payments_offered": 0,
               "out_msatoshi_offered": 0,
               "out_offered_msat": "0msat",
               "out_payments_fulfilled": 0,
               "out_msatoshi_fulfilled": 0,
               "out_fulfilled_msat": "0msat",
               "htlcs": []
            }
         ]
      },
      {
         "id": "02d77443ec4cb715251061b29be98cb347af453d5961227e44b7cfb6458d477331",
         "connected": true,
         "netaddr": [
            "10.71.230.231:58958"
         ],
         "globalfeatures": "",
         "localfeatures": "02aaa2",
         "features": "02aaa2",
         "channels": [
            {
               "state": "CHANNELD_NORMAL",
               "scratch_txid": "8fe7ec7f26ec29137f039adf73de9ed197f19d414b199dbe3e3918549fc0441c",
               "owner": "channeld",
               "short_channel_id": "221x1x0",
               "direction": 1,
               "channel_id": "c362119cff307b7d0b3d46515318d0bb2b55c7bd02fcda5ae06ed3d819e0d06e",
               "funding_txid": "6ed0e019d8d36ee05adafc02bdc7552bbbd0185351463d0b7d7b30ff9c1162c3",
               "close_to_addr": "bcrt1q7qya5p2kv5tax9aumnem8vda49a3gnffquh7jq",
               "close_to": "0014f009da05566517d317bcdcf3b3b1bda97b144d29",
               "private": false,
               "funding_allocation_msat": {
                  "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52": 0,
                  "02d77443ec4cb715251061b29be98cb347af453d5961227e44b7cfb6458d477331": 16777215000
               },
               "funding_msat": {
                  "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52": "0msat",
                  "02d77443ec4cb715251061b29be98cb347af453d5961227e44b7cfb6458d477331": "16777215000msat"
               },
               "msatoshi_to_us": 0,
               "to_us_msat": "0msat",
               "msatoshi_to_us_min": 0,
               "min_to_us_msat": "0msat",
               "msatoshi_to_us_max": 0,
               "max_to_us_msat": "0msat",
               "msatoshi_total": 16777215000,
               "total_msat": "16777215000msat",
               "dust_limit_satoshis": 546,
               "dust_limit_msat": "546000msat",
               "max_htlc_value_in_flight_msat": 18446744073709551615,
               "max_total_htlc_in_msat": "18446744073709551615msat",
               "their_channel_reserve_satoshis": 167772,
               "their_reserve_msat": "167772000msat",
               "our_channel_reserve_satoshis": 167772,
               "our_reserve_msat": "167772000msat",
               "spendable_msatoshi": 0,
               "spendable_msat": "0msat",
               "htlc_minimum_msat": 0,
               "minimum_htlc_in_msat": "0msat",
               "their_to_self_delay": 6,
               "our_to_self_delay": 6,
               "max_accepted_htlcs": 483,
               "status": [
                  "CHANNELD_NORMAL:Reconnected, and reestablished.",
                  "CHANNELD_NORMAL:Funding transaction locked. Channel announced."
               ],
               "in_payments_offered": 5,
               "in_msatoshi_offered": 400014,
               "in_offered_msat": "400014msat",
               "in_payments_fulfilled": 0,
               "in_msatoshi_fulfilled": 0,
               "in_fulfilled_msat": "0msat",
               "out_payments_offered": 0,
               "out_msatoshi_offered": 0,
               "out_offered_msat": "0msat",
               "out_payments_fulfilled": 0,
               "out_msatoshi_fulfilled": 0,
               "out_fulfilled_msat": "0msat",
               "htlcs": []
            }
         ]
      }
   ]
}
```

The paystatus should be checked on which node?
```
{
   "pay": [
      {
         "bolt11": "lnbcrt1u1p0rg0ptpp57d6npzjq8cpd4xrdxt8pgpcljhk436xvpz6yk2a9n233p8gux5jsdqjw3exzmnnv93hg6t0dcxqyjw5qcqp2sp5h4x26w8ntty29fpnt0ygrdzl5v096tyh9wsm9jdud86vjvuxfxts9qy9qsqqa74lkky6l3z2kfcj25gvlnqrfxne8zvacvkda6zavqmqx55e7jpvqva03kkf45mzdh8n4krw6eqx3lz2dupl5x3htvk0rmhnrvl8cgqquh6lx",
         "msatoshi": 100000,
         "amount_msat": "100000msat",
         "destination": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
         "attempts": [
            {
               "strategy": "Initial attempt",
               "start_time": "2020-01-31T14:39:39.154Z",
               "age_in_seconds": 4150,
               "end_time": "2020-01-31T14:39:39.380Z",
               "duration_in_seconds": 0,
               "route": [
                  {
                     "id": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
                     "channel": "221x1x0",
                     "direction": 0,
                     "msatoshi": 100002,
                     "amount_msat": "100002msat",
                     "delay": 16,
                     "style": "tlv"
                  },
                  {
                     "id": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
                     "channel": "209x1x0",
                     "direction": 1,
                     "msatoshi": 100000,
                     "amount_msat": "100000msat",
                     "delay": 10,
                     "style": "tlv"
                  }
               ],
               "failure": {
                  "code": 204,
                  "message": "Call to waitsendpay: failed: WIRE_TEMPORARY_CHANNEL_FAILURE (reply from remote)",
                  "data": {
                     "id": 5,
                     "payment_hash": "f375308a403e02da986d32ce14071f95ed58e8cc08b44b2ba59aa3109d1c3525",
                     "destination": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
                     "msatoshi": 100000,
                     "amount_msat": "100000msat",
                     "msatoshi_sent": 100002,
                     "amount_sent_msat": "100002msat",
                     "created_at": 1580481579,
                     "status": "pending",
                     "bolt11": "lnbcrt1u1p0rg0ptpp57d6npzjq8cpd4xrdxt8pgpcljhk436xvpz6yk2a9n233p8gux5jsdqjw3exzmnnv93hg6t0dcxqyjw5qcqp2sp5h4x26w8ntty29fpnt0ygrdzl5v096tyh9wsm9jdud86vjvuxfxts9qy9qsqqa74lkky6l3z2kfcj25gvlnqrfxne8zvacvkda6zavqmqx55e7jpvqva03kkf45mzdh8n4krw6eqx3lz2dupl5x3htvk0rmhnrvl8cgqquh6lx",
                     "erring_index": 1,
                     "failcode": 4103,
                     "failcodename": "WIRE_TEMPORARY_CHANNEL_FAILURE",
                     "erring_node": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
                     "erring_channel": "209x1x0",
                     "erring_direction": 1,
                     "raw_message": "1007008873c3d74eeca3fc047baf144584d2bc104eeaa9885f125c5584a84460cbbd304e310ad5355b43384e471c09fbbdd16b818e5748b728ecd62691b5a79cb2d9175106226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f0000d100000100005e34046e010100060000000000000000000000010000000a00000000ffffffff"
                  }
               }
            },
            {
               "strategy": "Excluded channel 209x1x0/1",
               "start_time": "2020-01-31T14:39:39.380Z",
               "age_in_seconds": 4150,
               "end_time": "2020-01-31T14:39:39.383Z",
               "duration_in_seconds": 0,
               "excluded_nodes_or_channels": [
                  "209x1x0/1"
               ],
               "failure": {
                  "code": 205,
                  "message": "Call to getroute: Could not find a route"
               }
            }
         ]
      }
   ]
}
```

**Comment by cdecker** [comment]
This is the `listpeers` output from the central node (`A` in your above description) through which the payment should be forwarded? In that case you do not have any outgoing capacity on either channel, and in particular you have no outgoing capacity on the edge `A -> C` (`0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52 -> 0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437` are the node IDs according to your information). You can see this by looking at the output of `listpeers` from A here:

```json
{
   "peers": [
      {
         "id": "0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437",
           ...
         "channels": [
            {
               ...
               "msatoshi_to_us": 0,
               "to_us_msat": "0msat",
               "spendable_msatoshi": 0,
               "spendable_msat": "0msat",
               ...
            }
         ]
      },
      {
         "id": "02d77443ec4cb715251061b29be98cb347af453d5961227e44b7cfb6458d477331",
         "connected": true,
           ...
         "channels": [
            {
               ...
               "msatoshi_to_us": 0,
               "to_us_msat": "0msat",
               "spendable_msatoshi": 0,
               "spendable_msat": "0msat",
               ...
            }
         ]
      }
   ]
}
```

You can also see this in the output of `paystatus` here:

```json
{
   "pay": [
      {
         "attempts": [
            {
               ...
               "failure": {
                  "code": 204,
                  "message": "Call to waitsendpay: failed: WIRE_TEMPORARY_CHANNEL_FAILURE (reply from remote)",
                  "data": {
                     "erring_index": 1,
                     "failcode": 4103,
                     "failcodename": "WIRE_TEMPORARY_CHANNEL_FAILURE",
                     "erring_node": "0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52",
                     "erring_channel": "209x1x0",
                  }
               }
            },
            {
               "strategy": "Excluded channel 209x1x0/1",
            }
         ]
      }
   ]
}
```
If you compare the channel being excluded it's the channel between `A` and `B` in this direction (`0331b7b4c1bb6352873db31ad19d5652c850ff52729207441c39e2ebc2358bbd52 -> 0308f128e28179cc910a344947d7acb6a5ab333eaa6cb8315745fb3aba56d5f437`) like we already deduced from the `listpeers` output.

TL;DR: `A` doesn't have any funds, so it can't forward anything.

**Comment by nachikettapas** [comment]
Thanks a lot @cdecker  for your patient explanation and time. I am including the additional steps and will let you know the result.

**Comment by nachikettapas** [comment]
I tried not with the node A having funds:
```
ubuntu@bitcoin-node-1:~$ lightning/cli/lightning-cli --network=regtest listfunds
{
   "outputs": [
      {
         "txid": "781c5c2de11bb3d5f65f23725b744856beaf2cc3b0c82762de2e3fd1be560398",
         "output": 1,
         "value": 100000000000,
         "amount_msat": "100000000000000msat",
         "address": "2NDfHBy7W7m775y39T4ByACqR21wZDB1A4r",
         "status": "confirmed",
         "blockheight": 203
      }
   ],
   "channels": [
      {
         "peer_id": "035bbd21ed423b01a4f7181e855e42a3e55b5ff24f02f2ec3a0c4fa12f55c6131c",
         "connected": true,
         "state": "CHANNELD_NORMAL",
         "short_channel_id": "411x1x0",
         "channel_sat": 10000,
         "our_amount_msat": "10000000msat",
         "channel_total_sat": 16777215,
         "amount_msat": "16777215000msat",
         "funding_txid": "17844fa177d8197be1e0d1d055c4de055eb99d83ef4823104444963e6993af29",
         "funding_output": 0
      },
      {
         "peer_id": "02463c962ccd088b2af77da619aff24859ff778dc30780d4ce25520cae9d582e31",
         "connected": true,
         "state": "CHANNELD_NORMAL",
         "short_channel_id": "423x1x0",
         "channel_sat": 5000,
         "our_amount_msat": "5000000msat",
         "channel_total_sat": 16777215,
         "amount_msat": "16777215000msat",
         "funding_txid": "4817185b8a68c69edd7a867c3152121206ceb75a2fe60dc5190e453e420b233c",
         "funding_output": 0
      }
   ]
}
```

Still a transaction from `02463c962ccd088b2af77da619aff24859ff778dc30780d4ce25520cae9d582e31 -> 02463c962ccd088b2af77da619aff24859ff778dc30780d4ce25520cae9d582e31` fails 

```
ubuntu@iot-1:~$ lightning/cli/lightning-cli --network=regtest paystatus
{
   "pay": [
      {
         "bolt11": "lnbcrt10p1p0rfzkppp5yq9m9p537fwu7e3j9902pttnyyg8xryqjxa8zssuyka3spsplghsdqjw3exzmnnv93hg6t0dcxqyjw5qcqp2sp53ffcvgpnhu0nr4ne5evjxsuywxy0493arka8lzjmn6xu22tjw4ns9qy9qsq4lj7jwzq0q50ye842khnqdvruttwj34tr9vsqg9h4hjmydhsnp0pvrw6ckx9tpk7hz9wdlyydcvt92h302cc0g50rt68gwgq86wptrgq9n9aeh",
         "msatoshi": 1,
         "amount_msat": "1msat",
         "destination": "035bbd21ed423b01a4f7181e855e42a3e55b5ff24f02f2ec3a0c4fa12f55c6131c",
         "shadow": "Added 6 cltv delay, 1 base fee, and 10 ppm fee for shadow to 0238829789bdfc4dc03ac8c279081e12a7b17996b54b2f47ce4e915d888bc139ad.Added 6 cltv delay, 1 base fee, and 10 ppm fee for shadow to 035bbd21ed423b01a4f7181e855e42a3e55b5ff24f02f2ec3a0c4fa12f55c6131c.Added 6 cltv delay, 1 base fee, and 10 ppm fee for shadow to 0238829789bdfc4dc03ac8c279081e12a7b17996b54b2f47ce4e915d888bc139ad.",
         "attempts": [
            {
               "strategy": "Initial attempt",
               "start_time": "2020-01-31T20:14:57.961Z",
               "age_in_seconds": 55,
               "end_time": "2020-01-31T20:14:58.190Z",
               "duration_in_seconds": 0,
               "route": [
                  {
                     "id": "0238829789bdfc4dc03ac8c279081e12a7b17996b54b2f47ce4e915d888bc139ad",
                     "channel": "423x1x0",
                     "direction": 1,
                     "msatoshi": 5,
                     "amount_msat": "5msat",
                     "delay": 34,
                     "style": "tlv"
                  },
                  {
                     "id": "035bbd21ed423b01a4f7181e855e42a3e55b5ff24f02f2ec3a0c4fa12f55c6131c",
                     "channel": "411x1x0",
                     "direction": 0,
                     "msatoshi": 4,
                     "amount_msat": "4msat",
                     "delay": 28,
                     "style": "tlv"
                  }
               ],
               "failure": {
                  "code": 204,
                  "message": "Call to waitsendpay: failed: WIRE_TEMPORARY_CHANNEL_FAILURE (reply from remote)",
                  "data": {
                     "id": 3,
                     "payment_hash": "200bb28691f25dcf6632295ea0ad732110730c8091ba71421c25bb180601fa2f",
                     "destination": "035bbd21ed423b01a4f7181e855e42a3e55b5ff24f02f2ec3a0c4fa12f55c6131c",
                     "msatoshi": 4,
                     "amount_msat": "4msat",
                     "msatoshi_sent": 5,
                     "amount_sent_msat": "5msat",
                     "created_at": 1580501697,
                     "status": "pending",
                     "bolt11": "lnbcrt10p1p0rfzkppp5yq9m9p537fwu7e3j9902pttnyyg8xryqjxa8zssuyka3spsplghsdqjw3exzmnnv93hg6t0dcxqyjw5qcqp2sp53ffcvgpnhu0nr4ne5evjxsuywxy0493arka8lzjmn6xu22tjw4ns9qy9qsq4lj7jwzq0q50ye842khnqdvruttwj34tr9vsqg9h4hjmydhsnp0pvrw6ckx9tpk7hz9wdlyydcvt92h302cc0g50rt68gwgq86wptrgq9n9aeh",
                     "erring_index": 1,
                     "failcode": 4103,
                     "failcodename": "WIRE_TEMPORARY_CHANNEL_FAILURE",
                     "erring_node": "0238829789bdfc4dc03ac8c279081e12a7b17996b54b2f47ce4e915d888bc139ad",
                     "erring_channel": "411x1x0",
                     "erring_direction": 0,
                     "raw_message": "100700885f96bd7e4f8749daf8b0a80f4606854686184b3deb324b35c06909019a099d48275312ed50417546972ca107ac69940dbbf08fef0f4ad0b57971a74aed1a4b4d06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f00019b00000100005e346808010000060000000000000000000000010000000a00000000ffffffff"
                  }
               }
            },
            {
               "strategy": "Excluded channel 411x1x0/0",
               "start_time": "2020-01-31T20:14:58.190Z",
               "age_in_seconds": 55,
               "end_time": "2020-01-31T20:14:58.193Z",
               "duration_in_seconds": 0,
               "excluded_nodes_or_channels": [
                  "411x1x0/0"
               ],
               "failure": {
                  "code": 205,
                  "message": "Call to getroute: Could not find a route"
               }
            }
         ]
      }
   ]
}
```

A noob question: does restarting a lightning node terminate a channel?

**Comment by ZmnSCPxj** [comment]
Hmmm.  We need the logs of 0238829789bdfc4dc03ac8c279081e12a7b17996b54b2f47ce4e915d888bc139ad (node A) then.

**Comment by ZmnSCPxj** [comment]
No, wait, you *still* have insufficient funds on the side of A:

```
         "our_amount_msat": "10000000msat",
         "channel_total_sat": 16777215,
```

This is because we need a "channel reserve" equal to 1% of the total channel capacity, which is a necessary security parameter of Lightning.  1% of 16777215 satoshi is 167,772.15 satoshi or 16,772,150millisatoshi.  Node A only has 10,000,000.  I suggest giving node A at least 167,774 satoshi (167,774,000 millisatoshi).

The channel reserve, as mentioned, is a necessary security parameter of Lightning: we allow channel to start without a reserve, but you can only receive if you have sent out at least the reserve amount.

**Comment by nachikettapas** [comment]
@ZmnSCPxj Thanks for the clarification. I have one more question. Is the channel uni-directional of bi-directional?

**Comment by ZmnSCPxj** [comment]
Channels are bidirectional.  However, it cannot send in a direction where all the funds already are.  As currently channels are single-funded, this implies that channels start *practically* unidirectional, in the direction from funder (opener) to fundee (acceptor).  As you send out, channels become practically bidirectional.

Dual-funding is in the works; this is more complicated since we must worry about privacy-leaking probe attacks that active channel openers can mount on passive public nodes.

**Comment by nachikettapas** [comment]
Well I tried multiple use-cases and then I stumbled across #3367. Closing it as a duplicate.

**Comment by ZmnSCPxj** [comment]
Huh? #3367 is `WIRE_INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS`, what you are showing here is `WIRE_TEMPORARY_CHANNEL_FAILURE`?

**Comment by nachikettapas** [comment]
Yes, initially the failure was due to `WIRE_TEMPORARY_CHANNEL_FAILURE` and I resolved it based on @cdecker  and your comments. However, payments were still failing randomly.

**Comment by ZmnSCPxj** [comment]
Ah, and then random failures were due to `WIRE_INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS` afterwards?  Okay.
